crypto/stack/stack.c

```diff
@@ -21,10 +28,43 @@ struct stack_st {
     OPENSSL_sk_compfunc comp;
 };
 
-#undef MIN_NODES
-#define MIN_NODES       4
+/*
+ * Calculate the array growth based on the target size.
+ *
+ * The growth faction is a rational number and is defined by a numerator
+ * and a denominator.  According to Andrew Koenig in his paper "Why Are
+ * Vectors Efficient?" from JOOP 11(5) 1998, this factor should be less
+ * than the golden ratio (1.618...).
+ *
+ * We use 3/2 = 1.5 for simplicty of calculation and overflow checking.
+ * Another option 8/5 = 1.6 allows for slightly faster growth, although safe
+ * computation is more difficult.
+ *
+ * The limit to avoid overflow is spot on.  2^n-1 is congruent to 0 mod 3
+ * when n is even and 1 mod 3 when n is odd.  When n is odd, int_max = 2^n-1
+ * produces limit = 2*floor(int_max/3) which is thus low by one, hence the
+ * correction term that adds one back for this case.
+ */
+static ossl_inline size_t compute_growth(size_t target, size_t current)
+{
+    const size_t limit = (INT_MAX / 3) * 2 + INT_MAX % 3;
+    size_t sz, n = current;
+
+    while (n <= target) {
+        /* This shouldn't be possible but check it anyway */
+        if (n >= INT_MAX)
```

Comment:
The commentary is misleading. Let's say `target` was INT_MAX. It should be legitimate value on 64-bit system [just legitimate, not practical]. But note that given `<=` loop condition and sz assignment below, it will be rejected. Once again, this is not about what's practical, but what's *formally* correct.

Reply:
This one shouldn't be possible because of the earlier checks that detect this case.
The loop can set `n` as high as INT_MAX and if that's insufficient the request would have been denied earlier.  I don't like the possibility of infinite loops however.

I'll improve the comment.

Reply:
In the loop body you have sz = ... INT_MAX, n = sz, then go to loop beginning, which is while (n <= target). If target is INT_MAX, it enters the loop and returns 0. When it should have returned INT_MAX.

> would have been denied earlier.

Earlier > INT_MAX is denied, not >= INT_MAX.

Reply:
I'll fix the `<=` check.  It should be '<'.